File: language/src/main/java/brj/nodes/StringNode.java
package brj.nodes;

import brj.BridjeLanguage;
import com.oracle.truffle.api.dsl.NodeField;
import com.oracle.truffle.api.dsl.Specialization;
import com.oracle.truffle.api.source.SourceSection;

@NodeField(name = "value", type = String.class)
@NodeField(name = "sourceSection", type = SourceSection.class)
public abstract class StringNode extends ExprNode {

    public StringNode(BridjeLanguage lang) {
        super(lang);
    }

    @Specialization
    public String doExecute(String value) {
        return value;
    }
}

File: language/src/main/kotlin/brj/nodes/StringNode.kt
package brj.nodes

import com.oracle.truffle.api.frame.VirtualFrame
import com.oracle.truffle.api.source.SourceSection

internal class StringNode(private val str: String,
                          override val loc: SourceSection?
) : ExprNode() {
    override fun execute(frame: VirtualFrame) = str
}